### ISSUE TITLE
감정 대화 분류 MZ CEC 해커톤 대회 (백업)

### ISSUE THREAD

**Issue by sallyy1** [sallyy1/NLU-NLP#1]
![다운로드](https://user-images.githubusercontent.com/70500841/101722021-af240b80-3aec-11eb-9551-5a96057c3126.jpeg)

<img width="1280" alt="1차 해커톤(딥러닝 기술 기반 감정 대화 분류) 과제 공지사항" src="https://user-images.githubusercontent.com/70500841/102463366-5c1efb00-408e-11eb-8815-e7d689945bf6.png">

- 주최측 제공 레퍼런스 : 
https://github.com/MahmoudWahdan/dialog-nlu/issues
https://github.com/MahmoudWahdan/dialog-nlu

- 사용한 Kobert :
https://github.com/monologg
https://github.com/huggingface/transformers

- 기타 참고 :
https://jybaek.tistory.com/781 (셔플)
https://lsjsj92.tistory.com/353 (꿈 많은 사람의 이야기)

https://huggingface.co/bert-base-cased
https://www.kaggle.com/kerneler/starter-bert-adamw-freezeembedding-dow-e4f2282c-6 (Freeze Embedding)
https://github.com/google-research/albert
https://www.kaggle.com/ykojima1989/fine-tune-bert-use-on-sklearn-pipeline

**Comment by sallyy1** [comment]
1) train data 특징 (Corpus 기준으로 추출했을 때)

[1] 토큰화 전
- 시퀀스 길이 : (평균: 167.7 / 최소: 37 / 최대: 495)
- (+ 사용자/시스템 단위 기준으로 추출했을 때, 31.2 / 2 / 125)

[2] 토큰화 적용 완료 후 (kobert_transformers 라이브러리의 get_tokenizer)
- 최대 길이: 291
- 최소 길이: 29



2) print(len(train_labels_large), len(validation_labels_large))
- 15832  / 1717
- train 데이터는 총 1만 5천 시퀀스

```
- TensorFlow 사용할 때에는
3) 원핫 인코딩
- 텐서플로우의 to_categorical 함수 사용
- (from tensorflow.keras.utils import to_categorical)
```

4) 모델
- 허깅페이스에서 kobert_transformer라이브러리 임포트하여 distilKobert 사용
- (from kobert_transformers import get_kobert_model, get_distilkobert_model)

5) 토그나이저
- 동일
- (from kobert_transformers import get_tokenizer)


6) max_len = 512 으로 시퀀스 길이 맞춰주기
- from tensorflow.keras.preprocessing.sequence import pad_sequences
- (train_padded = pad_sequences(train_sequences, maxlen=max_len, truncating=trunc_type, padding=padding_type))


7) loss는 크로스엔트로피(nn.CrossEntropyLoss()), 옵티마이저는 Radam/Adam


8) 기타
- batch_size 는 16으로 설정
- BERT-base(L=12, H=768, A=12) ->  L은 Transformer 블록의 숫자이고 H는 hidden size, A는 Transformer의 Attention block 숫자


9)  출력
- epoch : 7/20, batch : 650/989, small acc : 4644/10400 - (0.45), large acc : 6615/10400 - (0.64), loss : 0.154927

- 데이터 예
(사람) 진로가 너무 걱정스러워. 내가 진짜 뭘 해야 할지 모르겠어.
(시스템) 미래가 두려우시군요. 하고 싶은 일이 있으세요?
(사람) 하고 싶은 일은 있는데 지금 시작하기에 너무 늦은 것 같아서 선뜻 도전을 못 하겠어.
(시스템) 도전하는 것에 두려움을 느끼고 계시네요.
